c7n/actions.py

```diff
@@ -283,15 +304,28 @@ def get_groups(self, resources, metadata_key=None):
             elif remove_target_group_ids == 'all':
                 remove_groups = rgroups
             elif isinstance(remove_target_group_ids, list):
-                remove_groups = remove_target_group_ids
+                remove_groups = [r for r in remove_target_group_ids if r.startswith('sg-')]
+                remove_names = [r for r in remove_target_group_ids if not r.startswith('sg-')]
             elif isinstance(remove_target_group_ids, six.string_types):
-                remove_groups = [remove_target_group_ids]
+                if remove_target_group_ids.startswith('sg-'):
+                    remove_groups = [remove_target_group_ids]
+                else:
+                    remove_names = [remove_target_group_ids]
 
             # Parse add_groups
             if isinstance(add_target_group_ids, list):
-                add_groups = add_target_group_ids
+                add_groups = [a for a in add_target_group_ids if a.startswith('sg-')]
+                add_names = [a for a in add_target_group_ids if not a.startswith('sg-')]
             elif isinstance(add_target_group_ids, six.string_types):
-                add_groups = [add_target_group_ids]
+                if add_target_group_ids.startswith('sg-'):
+                    add_groups = [add_target_group_ids]
+                else:
+                    add_names = [add_target_group_ids]
+
+            if add_names:
+                add_groups += self.resolve_security_group_names(add_names)
```

Comment:
the initial values here were defined outside of the loop, so the list just grows with duplicate entries.